packages/mask/src/plugins/Pets/SNSAdaptor/RightMenu.tsx

```diff
@@ -0,0 +1,202 @@
+        animation: 'menu-show 0.5s both',
+        borderRadius: '12px',
+        pointerEvents: 'none',
+        boxShadow: '0 0 12px rgba(0,0,0,.1)',
+        '&>div': {
+            position: 'relative',
+            display: 'block',
+            padding: '20px',
+            fontSize: '14px',
+            transition: 'backgroundColor 100ms',
+            cursor: 'pointer',
+
+            '&>span': {
+                pointerEvents: 'none',
+            },
+            '&:hover': {
+                backgroundColor: '#f1f1f1',
+            },
+            '&:first-child': {
+                borderRadius: '12px 12px 0 0',
+            },
+            '&:last-child': {
+                borderRadius: '0 0 12px 12px',
+            },
+        },
+    },
+    show: {
+        opacity: '1 !important',
+        pointerEvents: 'auto',
+    },
+    menuItem: {
+        borderTop: '1px solid #f1f1f1',
+    },
+    icon: {
+        position: 'absolute',
+        top: '50%',
+        right: '20px',
+        transform: 'translateY(-50%)',
+        height: '16px',
+        width: 'auto',
+    },
+    itemBox: {
+        padding: '8px !important',
+        backgroundColor: 'transparent !important',
+        boxShadow: 'none !important',
+        '&>ul': {
+            display: 'block',
+            margin: 0,
+            padding: '20px',
+            fontSize: '12px',
+            color: '#444',
+            boxSizing: 'border-box',
+            boxShadow: '0 0 8px rgba(0,0,0,.1)',
+            fontWeight: 500,
+            backgroundColor: '#fff',
+            borderRadius: '12px',
+            position: 'absolute',
+            bottom: 0,
+            left: '8px',
+            '&>li': {
+                listStyle: 'none',
+                padding: '4px 0',
+            },
+        },
+    },
+    link: {
+        color: '#ccc',
+        transition: 'color 200ms',
+        '&:hover': {
+            color: '#444',
+        },
+    },
+    normal: {
+        fontWeight: 'normal',
+        color: '#777',
+    },
+}))
+
+function RightMenu(props: Props) {
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const refMenuDom = useRef<HTMLDivElement>(null)
+    const visitor = useCurrentVisitingUser(0)
+    const whoAmI = useLastRecognizedIdentity()
+
+    const [isLeft, setIsLeft] = useState(false)
+    const [isTop, setIsTop] = useState(false)
+
+    const { openDialog } = useRemoteControlledDialog(PluginPetMessages.events.essayDialogUpdated)
+    const { setDialog: openGameDialog } = useRemoteControlledDialog(PluginGameMessages.events.gameDialogUpdated)
+
+    function addEvents() {
+        document.body.addEventListener('click', props.onClose, false)
+        document.body.addEventListener('scroll', props.onClose, false)
+    }
+
+    function removeEvents() {
+        document.body.removeEventListener('click', props.onClose, false)
+        document.body.removeEventListener('scroll', props.onClose, false)
+    }
```

Comment:
Move both these two functions inside the useEffect hook bellow.(Don't forget to append `props.onClose` to its dependency array)